k8s/upgrades/0.9.0-1.0.0/jiva/jiva_volume_upgrade_auto.sh

```diff
@@ -0,0 +1,252 @@
+pv=$1
+upgrade_task=$2
+export pv
+export upgrade_task
+
+# Check if pv exists
+get_status=$(kubectl get pv "$pv" 2>&1);check_pv=$?
+if [ $check_pv -ne 0 ]; then
+    reason=$(echo $get_status | tr --delete ":")
+    patch_upgrade_task_error "$upgrade_task" "PRE_UPGRADE" "failed to upgrade $pv" "$reason"; 
+    exit 1
+fi
+
+# Check if CASType is jiva
+cas_type=$(kubectl get pv "$pv" -o jsonpath="{.metadata.annotations.openebs\.io/cas-type}")
+if [ "$cas_type" != "jiva" ]; then
+    reason=$(echo $get_status | tr --delete ":")
+    patch_upgrade_task_error "$upgrade_task" "PRE_UPGRADE" "failed to upgrade $pv" "invalid castype $cas_type"; 
+    exit 1
+fi
+
+ns=$(kubectl get pv "$pv" -o jsonpath="{.spec.claimRef.namespace}")
+export ns
+
+#pre checks for replica
+
+r_deploy_name=$(kubectl get deploy -n "$ns" \
+        -l openebs.io/persistent-volume="$pv",openebs.io/replica=jiva-replica \
+        -o jsonpath="{.items[*].metadata.name}" 2>&1 )
+rc=$?
+if [[ $rc -ne 0 ]]; then
+    reason=$(echo $r_deploy_name | tr --delete ":")
+    patch_upgrade_task_error "$upgrade_task" "PRE_UPGRADE" "failed to upgrade $pv" "$reason"; 
+    exit 1
+fi
+if [[ -z $r_deploy_name ]]; then
+    patch_upgrade_task_error "$upgrade_task" "PRE_UPGRADE" "failed to upgrade $pv" "replica deployment not found"; 
+    exit 1 
+fi
+export r_deploy_name
+
+r_con_name=$(kubectl get deploy -n "$ns" "$r_deploy_name" \
+        -o jsonpath="{range .spec.template.spec.containers[*]}{@.name}{'\n'}{end}" \
+        | grep "rep-con" 2>&1)
+rc=$?
+if [[ $rc -ne 0 ]]; then
+    reason=$(echo $r_con_name | tr --delete ":")
+    patch_upgrade_task_error "$upgrade_task" "PRE_UPGRADE" "failed to upgrade $pv" "$reason"; 
+    exit 1
+fi
+if [[ -z $r_con_name ]]; then
+    patch_upgrade_task_error "$upgrade_task" "PRE_UPGRADE" "failed to upgrade $pv" "replica conatiner not found"; 
+    exit 1 
+fi
+export r_con_name
+
+r_rs_old_list=$(kubectl get rs -o name --namespace "$ns" \
+        -l openebs.io/persistent-volume="$pv",openebs.io/replica=jiva-replica \
+        -o jsonpath='{range .items[*]}{@.metadata.name}:{end}' 2>&1)
+rc=$?
+if [[ $rc -ne 0 ]]; then
+    reason=$(echo $r_rs_old_list | tr --delete ":")
+    patch_upgrade_task_error "$upgrade_task" "PRE_UPGRADE" "failed to upgrade $pv" "$reason"; 
+    exit 1
+fi
+for r_rs in $(echo "$r_rs_old_list" | tr ":" " "); do
+    if [[ -z $r_rs ]]; then
+        patch_upgrade_task_error "$upgrade_task" "PRE_UPGRADE" "failed to upgrade $pv" "replica replicaset not found"
+        exit 1
+    fi
+done
+export r_rs_old_list
+
+replica_version=$(kubectl get deployment "$r_deploy_name" -n "$ns" \
+        -o jsonpath='{.metadata.labels.openebs\.io/version}')
+if [[ "$replica_version" != "$current_version" ]] && \
+    [[ "$replica_version" != "$upgrade_version" ]]; then
+    patch_upgrade_task_error "$upgrade_task" "PRE_UPGRADE" "failed to upgrade $pv" "Current Replica deployment $r_deploy_name version is not $current_version or $upgrade_version"
```

Comment:
Can we log current replica version using `$replica_version`

Reply:
Done